mappings/net/minecraft/client/util/math/Vector4f.mapping

```diff
@@ -1,13 +1,30 @@
 CLASS net/minecraft/class_1162 net/minecraft/client/util/math/Vector4f
+	FIELD field_21486 x F
+	FIELD field_21487 y F
+	FIELD field_21488 z F
+	FIELD field_21489 w F
 	METHOD <init> (FFFF)V
 		ARG 1 x
 		ARG 2 y
 		ARG 3 z
 		ARG 4 w
 	METHOD equals (Ljava/lang/Object;)Z
 		ARG 1 o
-	METHOD method_22674 multiply (Lnet/minecraft/class_1159;)V
+	METHOD method_22674 transform (Lnet/minecraft/class_1159;)V
+		ARG 1 function
+	METHOD method_23217 dotProduct (Lnet/minecraft/class_1162;)F
+		ARG 1 other
+	METHOD method_23218 normalize ()Z
+	METHOD method_23219 normalizeW ()V
+	METHOD method_23692 getColumn (ILnet/minecraft/class_1159;FFFF)F
```

Comment:
What does this do exactly? If it just gets a column from a `Matrix4f`, why is it in the `Vector4f` class, and why does it take so many parameters?

Reply:
It is a private method used for applying the matrix transformation (multiplying matrices).

Reply:
So it doesn't just get a column? It should have a better name in that case.

Reply:
this is more like calculate a new column. int is target row in matrix, the matrix is the matrix, and 4 floats are the original columns in the vector 4f (assuming the vector 4f is a 1 row by 4 column matrix)

Reply:
Then I think it should be something like `getMatrixProductColumn`. `getColumn` is too vague.

I'm still confused though, the result of multiplying a matrix by a vector is a vector, so why does it have several columns?